Cython/Utility/AsyncGen.c

```diff
@@ -1228,43 +1228,48 @@ __Pyx_async_gen_athrow_new(__pyx_PyAsyncGenObject *gen, PyObject *args)
 /* ---------- global type sharing ------------ */
 
 static int __pyx_AsyncGen_init(PyObject *module) {
+    PyTypeObject *type;
 #if CYTHON_USE_TYPE_SPECS
-    __pyx_AsyncGenType = __Pyx_FetchCommonTypeFromSpec(module, &__pyx_AsyncGenType_spec, NULL);
+    type = __Pyx_FetchCommonTypeFromSpec(module, &__pyx_AsyncGenType_spec, NULL);
 #else
     (void) module;
     // on Windows, C-API functions can't be used in slots statically
     __pyx_AsyncGenType_type.tp_getattro = __Pyx_PyObject_GenericGetAttrNoDict;
-    __pyx_AsyncGenType = __Pyx_FetchCommonType(&__pyx_AsyncGenType_type);
+    type = __Pyx_FetchCommonType(&__pyx_AsyncGenType_type);
 #endif
-    if (unlikely(!__pyx_AsyncGenType))
+    if (unlikely(!type))
         return -1;
+    __Pyx_CGlobal(AsyncGenType) = type;
```

Comment:
This looks so wrong in the C code, there should really be a setter macro which does this. Even if it's not used globally for now (pun not intended), it would avoid having to write code by hand that assigns to the result of a "function call".

Reply:
It looked less wrong when it was still `Globals()->AsynGenType = ...`. I would still like to postpone the getters/setter functions, things are already tricky to get working in the current state where I do not need to review every global access.